pkg/client/openstack/admin/client.go

```diff
@@ -251,3 +260,54 @@ func (c *adminClient) getRoleID(roleName string) (string, error) {
 	return "", fmt.Errorf("Role %s not found", roleName)
 
 }
+
+func (c *adminClient) CreateContainer(containerName, serviceUserName, serviceUserDomainName, serviceUserProjectID string) error {
+	listOpts := endpoints.ListOpts{
+		Availability: "public",
+	}
+
+	allPages, err := endpoints.List(c.IdentityClient, listOpts).AllPages()
+	if err != nil {
+		return err
+	}
+
+	allEndpoints, err := endpoints.ExtractEndpoints(allPages)
+	if err != nil {
+		return err
+	}
+
+	var swiftEndpoint endpoints.Endpoint
+	for _, endpoint := range allEndpoints {
+		if endpoint.Name == "swift" {
```

Comment:
I would filter by `type` instead. The name is arbitrarily choosen.
```suggestion
		if endpoint.Type == "objectstore" {
```

Reply:
@databus23 It seems that Endpoint.Type is not available, see https://github.com/gophercloud/gophercloud/blob/master/openstack/identity/v3/endpoints/results.go#L41